src/data/bucket/symbol_bucket.js

```diff
@@ -414,13 +438,34 @@ class SymbolBucket implements Bucket {
             this.symbolInstances = options.symbolInstances;
 
             const layout = options.layers[0].layout;
-            this.sortFeaturesByY = layout['text-allow-overlap'] || layout['icon-allow-overlap'] ||
-                layout['text-ignore-placement'] || layout['icon-ignore-placement'];
+            this.sortFeaturesByY = layout.get('text-allow-overlap') || layout.get('icon-allow-overlap') ||
+                layout.get('text-ignore-placement') || layout.get('icon-ignore-placement');
 
         } else {
-            const layer = this.layers[0];
-            this.textSizeData = getSizeData(this.zoom, layer, 'text-size');
-            this.iconSizeData = getSizeData(this.zoom, layer, 'icon-size');
+            const layer: SymbolStyleLayer = this.layers[0];
+            const unevaluatedLayoutValues = layer._unevaluatedLayout._values;
```

Comment:
Oh, is this because `getLayoutProperty`'s return type isn't parameterized on the property name?

Reply:
It's because `layer.getLayoutProperty` returns the raw style spec value (as needed for the public runtime styling API), and we need the `PropertyValue`.

Reply:
Gotcha.  I guess we could add `SymbolStyleLayer#get{Text,Icon}Size()`... but allowing treating *_bucket as a 'friend' of *_style_layer is probably reasonable, too.